x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -93,22 +89,17 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
             return;
         }
 
-        if (extractClientCertificate) {
-            HttpChannel httpChannel = request.getHttpChannel();
-            SSLEngineUtils.extractClientCertificates(logger, threadContext, httpChannel);
-        }
-
-        authenticationService.authenticate(maybeWrapRestRequest(request), ActionListener.wrap(authentication -> {
+        final HttpRequest httpRequest = wrapHttpContent(request.getHttpRequest(), request.getXContentType());
+        authenticationService.authenticate(httpRequest, ActionListener.wrap(authentication -> {
             if (authentication == null) {
-                logger.trace("No authentication available for REST request [{}]", request.uri());
+                logger.trace("No authentication available for REST request [{}]", httpRequest.uri());
             } else {
-                logger.trace("Authenticated REST request [{}] as {}", request.uri(), authentication);
+                logger.trace("Authenticated REST request [{}] as {}", httpRequest.uri(), authentication);
             }
-            secondaryAuthenticator.authenticateAndAttachToContext(request, ActionListener.wrap(secondaryAuthentication -> {
+            secondaryAuthenticator.authenticateAndAttachToContext(httpRequest, ActionListener.wrap(secondaryAuthentication -> {
                 if (secondaryAuthentication != null) {
-                    logger.trace("Found secondary authentication {} in REST request [{}]", secondaryAuthentication, request.uri());
+                    logger.trace("Found secondary authentication {} in REST request [{}]", secondaryAuthentication, httpRequest.uri());
                 }
-                RemoteHostHeader.process(request, threadContext);
```

Comment:
This has also been moved to the consolidated thread context populator, **before** running authentication (here it is "after", but moving it "before" allows more audit events to benefit, and aids the transion from `RestRequest` to `BasicHttpRequest` in the auditing service).